cmd/secrets/secret_helper.go

```diff
@@ -37,14 +56,23 @@ var SecretHelperCmd = &cobra.Command{
 	Long:  ``,
 }
 
-// ReadSecretsCmd implements reading secrets from a directory/volume mount
 var readSecretCmd = &cobra.Command{
 	Use:   "read",
-	Short: "Read secret from a directory",
+	Short: "Read secrets",
 	Long:  ``,
-	Args:  cobra.ExactArgs(1),
+	Args:  cobra.MaximumNArgs(1), // 0 when using the backend prefixes option, 1 when reading a file
 	RunE: func(cmd *cobra.Command, args []string) error {
-		return readSecrets(os.Stdin, os.Stdout, args[0])
+		usePrefixes, err := cmd.Flags().GetBool(backendPrefixesFlag)
+		if err != nil {
+			return err
+		}
+
+		dir := ""
+		if len(args) == 1 {
+			dir = args[0]
+		}
+
+		return decodeSecrets(os.Stdin, os.Stdout, dir, usePrefixes, &backendSelector{})
```

Comment:
nit: I think the old function name might have been clearer - we're not specifically decoding the secrets but fetching/reading/getting them.